### PR TITLE
porting the dithering function from kaldi

### PR DESCRIPTION
**Pull request by KarelVesely84**

### CODE REVIEW

File: kaldi-native-fbank/csrc/kaldi-math.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#ifndef M_2PI
+#define M_2PI 6.283185307179586476925286766559005
+#endif
+
+namespace knf {
+
+inline float Log(float x) { return logf(x); }
+
+// Returns a random integer between 0 and RAND_MAX, inclusive
+int Rand(struct RandomState* state = NULL);
```
Comment (csukuangfj) [inline kaldi-native-fbank/csrc/kaldi-math.h:25]: By the way, could you upload `kaldi-math.cc`? The `.cc` file contains the implementation for this function.

Comment (KarelVesely84) [reply kaldi-native-fbank/csrc/kaldi-math.h:25]: strange, I did not port it, and yet it compiled locally...

Comment (csukuangfj) [reply kaldi-native-fbank/csrc/kaldi-math.h:25]: CI failed
https://github.com/csukuangfj/kaldi-native-fbank/actions/runs/8268120256/job/22640066618?pr=40#step:6:23
```
[ 40%] Building CXX object kaldi-native-fbank/csrc/CMakeFiles/test-online-fbank.dir/test-online-fbank.cc.o
[ 44%] Linking CXX executable ../../bin/test-online-fbank
/usr/bin/ld: ../../lib/libkaldi-native-fbank-core.so: undefined reference to `knf::Rand(knf::RandomState*)'
/usr/bin/ld: ../../lib/libkaldi-native-fbank-core.so: undefined reference to `knf::RandomState::RandomState()'
collect2: error: ld returned 1 exit status
make[2]: *** [kaldi-native-fbank/csrc/CMakeFiles/test-online-fbank.dir/build.make:98: bin/test-online-fbank] Error 1
make[1]: *** [CMakeFiles/Makefile2:369: kaldi-native-fbank/csrc/CMakeFiles/test-online-fbank.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 48%] Linking CXX shared library ../../../lib/libgtest.so
[ 48%] Built target gtest
make: *** [Makefile:146: all] Error 2
Error: Process completed with exit code 2.
```

By the way, you also need to add `kaldi-math.cc` to CMakeLists.txt

**Comment by KarelVesely84** [comment]
okay, I added the missing part

**Comment by KarelVesely84** [comment]
for me it is ready, i just changed the default dithering constant

k2 uses audio samples [-1..+1]
kaldi was using audio samples [-32k..+32k], hence the constant is 2^15x smaller

**Comment by csukuangfj** [comment]
Thanks! Will check it tomorrow morning.

**Comment by csukuangfj** [comment]
Thank you for your contribution!

**Comment by nshmyrev** [comment]
Hey, can we please change dither constant to be 1.0 like it was before. 0.00003 vs 1.0 can cause a lot of confusion in existing code.

**Comment by csukuangfj** [comment]
Could you create a PR to change it to 1.0?

**Comment by nshmyrev** [comment]
Let it stay then, sorry for confusion and thanks for the patch.

**Pull request merged**